server/server.go

```diff
@@ -825,3 +772,34 @@ func (server HTTPServer) getOverviewPerCluster(
 		TagsHit:       tags,
 	}, nil
 }
+
+// filterRulesResponse returns an array of RuleWithContentResponse with only the rules that matches 3 criteria:
+// - The rule has content from the content-service
+// - The disabled filter is not match
+// - The OSD elegible filter is not match
+func filterRulesResponse(aggregatorReport []types.RuleOnReport, filterOSD, filterDisabled bool) (
+	filteredRules []proxy_types.RuleWithContentResponse,
+	noContentRules int,
+) {
+	filteredRules = []proxy_types.RuleWithContentResponse{}
+	noContentRules = 0
+
+	for _, aggregatorRule := range aggregatorReport {
+		if aggregatorRule.Disabled && filterDisabled {
+			continue
+		}
+
+		rule, successful, filtered := content.FetchRuleContent(aggregatorRule, filterOSD)
```

Comment:
are we blindly "forget" rules w/o content there?

Reply:
The next `if` block checks it and increase the `noContentRules` variable, which is one of the returned ones. If I'm not mistaken, the code behaves exactly as the previous version (which probably needs a refactor, maybe as a different task)